bot/cogs/moderation/incidents.py

```diff
@@ -0,0 +1,334 @@
+    Add `Signal` member emoji to `incident` as reactions.
+
+    If the emoji has already been placed on `incident` by the bot, it will be skipped.
+    """
+    existing_reacts = own_reactions(incident)
+
+    for signal_emoji in Signal:
+
+        # This will not raise, but it is a superfluous API call that can be avoided
+        if signal_emoji.value in existing_reacts:
+            log.trace(f"Skipping emoji as it's already been placed: {signal_emoji}")
+
+        else:
+            log.trace(f"Adding reaction: {signal_emoji}")
+            await incident.add_reaction(signal_emoji.value)
+
+
+class Incidents(Cog):
+    """
+    Automation for the #incidents channel.
+
+    This cog does not provide a command API, it only reacts to the following events.
+
+    On start-up:
+        * Crawl #incidents and add missing `Signal` emoji where appropriate
+        * This is to retro-actively add the available options for messages which
+          were sent while the bot wasn't listening
+        * Pinned messages and message starting with # do not qualify as incidents
+        * See: `crawl_incidents`
+
+    On message:
+        * Add `Signal` member emoji if message qualifies as an incident
+        * Ignore messages starting with #
+            * Use this if verbal communication is necessary
+            * Each such message must be deleted manually once appropriate
+        * See: `on_message`
+
+    On reaction:
+        * Remove reaction if not permitted (`ALLOWED_EMOJI`, `ALLOWED_ROLES`)
+        * If `Signal.ACTIONED` or `Signal.NOT_ACTIONED` were chosen, attempt to
+          relay the incident message to #incidents-archive
+        * If relay successful, delete original message
+        * See: `on_raw_reaction_add`
+
+    Please refer to function docstrings for implementation details.
+    """
+
+    def __init__(self, bot: Bot) -> None:
+        """Prepare `event_lock` and schedule `crawl_task` on start-up."""
+        self.bot = bot
+
+        self.event_lock = asyncio.Lock()
+        self.crawl_task = self.bot.loop.create_task(self.crawl_incidents())
+
+    async def crawl_incidents(self) -> None:
+        """
+        Crawl #incidents and add missing emoji where necessary.
+
+        This is to catch-up should an incident be reported while the bot wasn't listening.
+        After adding each reaction, we take a short break to avoid drowning in ratelimits.
+
+        Once this task is scheduled, listeners that change messages should await it.
+        The crawl assumes that the channel history doesn't change as we go over it.
+        """
+        await self.bot.wait_until_guild_available()
+        incidents: discord.TextChannel = self.bot.get_channel(Channels.incidents)
+
+        # Limit the query at 50 as in practice, there should never be this many messages,
+        # and if there are, something has likely gone very wrong
+        limit = 50
+
+        # Seconds to sleep after adding reactions to a message
+        sleep = 2
```

Comment:
Yeah, maybe, I'm not sure. I placed them here because I thought their purpose would be better understood when defined where they are used, especially since they are used in one place only, and because you don't really need to see these at the top of the module. In fact, the only reason why I defined them here rather than just passing them into `history` and `sleep` directly as literals is that I wanted to log them on line `L131` before starting the loop.

I'm not opposed to putting them at the top, but I'd prefer to hear at least one more opinion on it.

Reply:
I don't think these are necessarily constants worthy, putting them at the top of the file seems fine IMO. Constants are useful for development when testing things like the offensive message remover (we don't want to sit waiting for a week to confirm it works) but I think there is little reason to change these and they are constant... constants. Putting them at the top of the file would be nicer though I think.

Reply:
I made them module-level constants in 6d3a91c. I'm still not entirely convinced that it's better in any way (now they need more documentation), but I don't feel strongly about it.

Reply:
Perhaps class constants would be more fitting to give the names a bit more meaning without explaining it all